src/aks-preview/azext_aks_preview/_validators.py

```diff
@@ -389,10 +423,31 @@ def validate_max_surge(namespace):
         raise CLIError("--max-surge should be an int or percentage")
 
 
+def validate_addons(namespace):
+    if not hasattr(namespace, 'addons'):
+        return
+    addons = namespace.addons
+    addon_args = addons.split(',')
+
+    for addon_arg in addon_args:
+        if addon_arg not in ADDONS:
+            matches = _fuzzy_match(addon_arg, list(ADDONS))
+            matches = str(matches)[1:-1]
+            all_addons = list(ADDONS)
+            all_addons = str(all_addons)[1:-1]
+            if len(matches) == 0:
+                raise CLIError(
+                    f"The addon \"{addon_arg}\" is not a recognized addon option. Possible options: {all_addons}")
+            else:
```

Comment:
Nit: I don't think you need the else here. 

Because after a raise occurs the function should terminate afaik